src/Changes.php

```diff
@@ -2,55 +2,53 @@
 
 declare(strict_types=1);
 
-namespace Roave\ApiCompare;
+namespace Roave\BackwardCompatibility;
 
 use ArrayIterator;
-use Assert\Assert;
 use Countable;
 use IteratorAggregate;
-use function array_merge;
 use function count;
 
 final class Changes implements IteratorAggregate, Countable
 {
     /** @var Change[] */
-    private $changes = [];
+    private $changes;
 
     private function __construct()
     {
     }
 
-    public static function new() : self
+    public static function empty() : self
     {
-        return new self();
-    }
+        static $empty;
 
-    /**
-     * @param Change[] $changes
-     * @return Changes
-     */
-    public static function fromArray(array $changes) : self
-    {
-        Assert::that($changes)->all()->isInstanceOf(Change::class);
-        $instance          = self::new();
-        $instance->changes = $changes;
-        return $instance;
+        if ($empty) {
```

Comment:
I found (via `Kcachegrind`) that this was really trashing memory and CPU on this side (together with `fromArray()` and `mergeWith()`)